chrome/content/coniks/vrf.js

```diff
@@ -0,0 +1,32 @@
+this.EXPORTED_SYMBOLS = ["VRF"];
+
+var { interfaces: Ci, utils: Cu, classes: Cc } = Components;
+
+Cu.import("resource://gre/modules/ctypes.jsm");
+Cu.import("chrome://otr/content/helpers.js");
+Cu.import("chrome://otr/content/coniks/libconiks.js");
+Cu.import("chrome://otr/content/coniks/util.js");
+
+var VRF = {
+  /**
+   * @param {Array} pk - VRF public key.
+   * @param {string} m - The message.
+   * @param {Array} index - VRF index.
+   * @param {Array} proof - VRF proof.
+   */
+  verify: function(pk, m, index, proof) {
+    let msg = util.stringToArray(m);
+
+    let pkPtr = ctypes.unsigned_char.array()(pk);
+    let msgPtr = ctypes.unsigned_char.array()(m);
+    let indexPtr = ctypes.unsigned_char.array()(index);
+    let proofPtr = ctypes.unsigned_char.array()(proof);
+    if (libCONIKS.cgoVerifyVrf(pkPtr, pk.length,
```

Comment:
I thought it might provide an extra check that on the JS side everything works as expected. It might be an overkill as the cgo part is already (well) tested. The second sentence refers to the `FromBytesBaseGroup` in the coname's vrf function (which lacks some checks).